server/service/hosts.go

```diff
@@ -1789,16 +1796,85 @@ func (svc *Service) OSVersions(ctx context.Context, teamID *uint, platform *stri
 			// most of the time, team should exist so checking here saves unnecessary db calls
 			_, err := svc.ds.Team(ctx, *teamID)
 			if err != nil {
-				return nil, err
+				return nil, count, nil, err
 			}
 		}
 		// if team exists but stats have not yet been gathered, return empty JSON array
 		osVersions = &fleet.OSVersions{}
 	} else if err != nil {
-		return nil, err
+		return nil, count, nil, err
+	}
+
+	for i, os := range osVersions.OSVersions {
+
+		// populate OSVersion.GeneratedCPEs
+		if os.Platform == "darwin" {
+			osVersions.OSVersions[i].GeneratedCPEs = []string{
+				fmt.Sprintf("cpe:2.3:o:apple:macos:%s:*:*:*:*:*:*:*", os.Version),
+				fmt.Sprintf("cpe:2.3:o:apple:mac_os_x:%s:*:*:*:*:*:*:*", os.Version),
+			}
+		}
+
+		// populate OSVersion.Vulnerabilities
+		vulns, err := svc.ds.ListVulnsByOS(ctx, os.ID, false)
+		if err != nil {
+			return nil, count, nil, err
+		}
+
+		osVersions.OSVersions[i].Vulnerabilities = make(fleet.Vulnerabilities, 0) // avoid null in JSON
+		for _, vuln := range vulns {
+			switch os.Platform {
+			case "darwin":
+				vuln.DetailsLink = fmt.Sprintf("https://nvd.nist.gov/vuln/detail/%s", vuln.CVE)
+			case "windows":
+				vuln.DetailsLink = fmt.Sprintf("https://msrc.microsoft.com/update-guide/en-US/vulnerability/%s", vuln.CVE)
+			}
+			osVersions.OSVersions[i].Vulnerabilities = append(osVersions.OSVersions[i].Vulnerabilities, vuln)
+		}
+	}
+
+	if opts.OrderKey == "hosts_count" && opts.OrderDirection == fleet.OrderAscending {
```

Comment:
I see a lot of similarities between this and the ee version. Can they share some of the code in this method?